CAIPs/caip-stream-sync.md

```diff
@@ -0,0 +1,193 @@
+5. `B` registers that `A` is subscribed to `s` with a ttl of 12h
+6. `B` sends a `HaveTips` message with the tips it has validated as a response to `A`
+7. `A` registers that `B` is subscribed to `s'` (as defined in the `HaveTips` message) with a ttl of 12h
+
+#### Publishing updates
+
+1. A new event is created in a stream of `s`, the event creator adds this event to peer `A`
+2. `A` sends a `Have` message to all peers in its registry which subscribe to `s`
+
+#### Stream events
+
+1. `A` sends a `WantStream` message to `B` based on `A`s current knowledge of `s`
+2. `B` responds with a `StreamChunk` message
+3. `A` repeats step 1 until it has caught up with the desired tip(s)
+
+#### Maintaining connectivity
+
+1. Every 6h nodes will send new `SubscribeSet` messages for all of its streams to all connected peers
+1. Recieving peers will respond as described in *Tip Subscription* from step 5
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+At a first glance the `SubscribeSet` and `HaveTips` messages and protocol one might consider why not to use libp2p gossipsub instead. While gossipsub is a great protocol that has been battle tested there are a few reasons why it is not optimal for this use case. 
+
+First of all it forces all peers to agree on a particular pubsub topic. For example a topic based on the *separatorKey/Value*. This would force all nodes to listen to all events published to this stream set. With the protocol defined above peers can add additional constraints to the updates that they care about by specifying the *stream-* and *controllerFilter*. In practice this means that node size can be heterogeneous, i.e. some large nodes that subscribe to everything and many smaller nodes that only subscribe to some subset of streams. 
+
+Another consideration is that in gossipsub messages are propagated though many peers. However, in order to synchronize the events of a stream peers need to be directly connected to the originating peer anyway. In the future peers might be disincentivized to forward `HaveTips` messages since future versions of this protocol could introduce payments for event retrieval.
+
+The purpose of this protocol is to synchronize hash linked and signed event logs. This is the reason that event streams need to be synchronized in verification order, e.g. `event0` -> `event1` -> `event2` and so on. This is to prevent DoS attacks where if the events is sent from tip to first event an attacker can sent a lot of invalid events and the victim node would need to synchronize all data before seeing that these events are invalid.
```

Comment:
```suggestion
The purpose of this protocol is to synchronize hash linked and signed event logs. This is the reason that event streams need to be synchronized in verification order, e.g. `event0` -> `event1` -> `event2` and so on. This is to prevent DoS attacks where if the events are sent from tip to first event an attacker can send a lot of invalid events and the victim node would need to synchronize all data before seeing that these events are invalid.
```